include/mbedtls/ssl.h

```diff
@@ -1784,15 +1784,24 @@ void mbedtls_ssl_conf_sig_hashes( mbedtls_ssl_config *conf,
 
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
 /**
- * \brief          Set the hostname to check against the received server
- *                 certificate. It sets the ServerName TLS extension too,
- *                 if the extension is enabled.
- *                 (client-side only)
+ * \brief          Set or reset the hostname to check against the received 
+ *                 server certificate. It sets the ServerName TLS extension, 
+ *                 too, if that extension is enabled. (client-side only)
  *
  * \param ssl      SSL context
- * \param hostname the server hostname
+ * \param hostname the server hostname, may be NULL to clear hostname
+ 
+ * \note           Maximum hostname length MBEDTLS_SSL_MAX_HOST_NAME_LEN.
+ *
+ * \return         0 if successful, MBEDTLS_ERR_SSL_ALLOC_FAILED on 
+ *                 allocation failure, MBEDTLS_ERR_SSL_BAD_INPUT_DATA on 
+ *                 too long input hostname.
+ *
+ *                 Hostname is set to the one provided if successful (cleared
+ *                 when NULL). On allocation failure MBEDTLS_ERR_SSL_ALLOC_FAILED, 
+ *                 hostname is cleared. On too long input failure
+ *                 MBEDTLS_ERR_SSL_BAD_INPUT_DATA, old hostname is unchanged.
```

Comment:
They repeat the error code, but the essential (new) information is to describe if and how the hostname has changed on a failure.